diff --git a/nix/default.nix b/nix/default.nix
index 51bb58fe..5dda1a82 100644
--- a/nix/default.nix
+++ b/nix/default.nix
@@ -36,6 +36,7 @@
   libxkbcommon,
   libuuid,
   libgbm,
+  lua54,
   muparser,
   pango,
   pciutils,
@@ -190,6 +191,7 @@ customStdenv.mkDerivation (finalAttrs: {
       libuuid
       libxcursor
       libxkbcommon
+      lua54
       muparser
       pango
       pciutils
diff --git a/nix/overlays.nix b/nix/overlays.nix
index 0d157701..c538dc70 100644
--- a/nix/overlays.nix
+++ b/nix/overlays.nix
@@ -49,6 +49,7 @@ in
       {
         hyprland = final.callPackage ./default.nix {
           stdenv = final.gcc15Stdenv;
+          lua54 = final.lua54Packages.lua; # lua54Packages.lua doesn't work in build
           commit = self.rev or "";
           revCount = self.sourceInfo.revCount or "";
           inherit date version;
diff --git a/src/config/supplementary/jeremy/Jeremy.hpp b/src/config/supplementary/jeremy/Jeremy.hpp
index c1332c28..5ea82a03 100644
--- a/src/config/supplementary/jeremy/Jeremy.hpp
+++ b/src/config/supplementary/jeremy/Jeremy.hpp
@@ -2,6 +2,7 @@
 
 #include <string>
 #include <expected>
+#include <cstdint>
 
 namespace Config::Supplementary::Jeremy {
     enum eConfigType : uint8_t {
